---
title: "Ch1"
author: "Madison Chapel"
date: "2021-09-03"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# 1. Generative Models for Discrete Data

### 1.2 example

Display how often 3 mutations would occur in the Poisson(5) model: 

```{r dpois2}
dpois(x = 3, lambda = 5)
```

Okay, I think I've got this RMarkdown business (partially) figured out! 

We can also vector our probabilities for all values 0 to 12:

``` {r dpois3}
dpois(x = 0:12, lambda = 5)
```

``` {r dpois4, echo =FALSE}
barplot(dpois(0:12,5), names.arg = 0:12, col = "purple")
```


### 1.3 Discrete probability models

**Q 1.1**
What if you want to create a factor that has some levels not yet in your data?

**Answer**
```{r 1.1}
genotype = c("AA","AO","BB","AO","OO")

genotypeF = factor(c(genotype,"Level"))

levels(genotypeF)
```

#### 1.3.1 Bernoulli trials 

**Q 1.2**
Repeat this function call a number of times. Why isn't the answer always the same? 

```{r 1.2}
rbinom(15, prob = 0.5, size = 1)

rbinom(15, prob = 0.5, size = 1)

rbinom(15, prob = 0.5, size = 1)
```

Just because the probability of success is 0.5 doesn't mean that half the events will be 1 and half will be 0. The 0.5 probability applies to each event individually, which is why you get different outcomes every time you repeat several trials. 

#### 1.3.2 Binomial success counts

If we only care about successes, we can set `size` to the total number of trials: 

``` {r 1.3.2}
rbinom(1, prob = 2/3, size = 12)
```

**Q 1.3** 
Repeat this function call ten times. What seems to be the most common outcome? 

```{r 1.3}
rbinom(1, prob = 0.3, size = 15)
rbinom(1, prob = 0.3, size = 15)
rbinom(1, prob = 0.3, size = 15)
rbinom(1, prob = 0.3, size = 15)
rbinom(1, prob = 0.3, size = 15)
rbinom(1, prob = 0.3, size = 15)
rbinom(1, prob = 0.3, size = 15)
rbinom(1, prob = 0.3, size = 15)
rbinom(1, prob = 0.3, size = 15)
rbinom(1, prob = 0.3, size = 15)
```
An outcome of 6 seems fairly common, but this will change from one set of 10 to another. A sample size of 10 isn't actually big enough to see what the most common outcome really is. Looking at the distribution is a better way: 

``` {r 1.3fig, echo = FALSE}

probabilities = dbinom(0:15, prob = 0.3, size = 15)
barplot(probabilities, names.arg = 0:15, col = "purple")
```
 
The actual most frequent value will be 4!  


**Q 1.4**
What is the output of the formula for k = 3, p = 2/3, n = 4? 

``` {r 1.4} 
probabilities = dbinom(0:4, prob = 2/3, size = 4)
round(probabilities, 4)

```

The probability of seeing X = 3 successes is 0.3951. Can I check this with the formula from the text, too? 

``` {r 1.4math}
n = 4
k = 3
p = 2/3
P = ((factorial(n)/(factorial((n-k))*factorial(k))))*(p^k)*(1-p)^(n-k)
P
```

That's the same! Satisfying! 

#### 1.3.3. Poisson Distributions

When *p* is small and trials *n* is large, binomial distribution can be approximated with Poisson, with rate parameter *lambda = np*. 

**Q 1.5**
What is the probability mass distribution of observing 0:12 mutations in a genome of n = 10^4^ nucleotides, when the probability is p = 5 x 10^-4^ per nucleotide? 
Is it similar when modeled by the binomial distribution and the Poisson distribution? 

``` {r 1.5, echo = FALSE}

binom = dbinom(0:12, prob = 5*10^(-4), size = 10^4)
poisson = dpois(0:12, (5*10^(-4))*(10^4))

df <- data.frame(binom, poisson)

barplot(t(as.matrix(df)),
        beside = TRUE,
        names.arg = c(0:12),
        )


```

The probability distributions are very similar when modeled by binomial distribution and Poisson distribution, confirming that the Poisson distribution is a good approximation in certain situations. 

